### PR TITLE
do not run validation for empty fields

### PR DESCRIPTION
**Pull request by vasinkevych**

### CODE REVIEW

File: packages/core/src/utilities/validation.js [hunk 1]
```diff
@@ -314,9 +314,9 @@ export const validators = {
 
 export const hasValue = (value: Value): boolean => {
   const valueIsEmptyArray = Array.isArray(value) && value.length === 0;
-  const hasNoValue =
```
Comment (vasinkevych) [inline packages/core/src/utilities/validation.js]: I've add some tests below for this changes

File: packages/core/src/comparator.test.js [hunk 2]
```diff
@@ -146,12 +147,14 @@ describe("compare date fields", () => {
...
-    expect(form.state().fields[1].errorMessages).toBe(secondDateError);
+    expect(form.state().fields[0].errorMessages).toBe(
+      "A value must be provided"
+    );
+    expect(form.state().fields[1].errorMessages).toBe("");
   });
 
   test("warning shown for fields", () => {
-    expect(form.find("span.errors").length).toBe(2);
+    expect(form.find("span.errors").length).toBe(1);
```
Comment (vasinkevych) [inline packages/core/src/comparator.test.js]: - [ ]  check case and add test with one empty field

**Comment by vasinkevych** [comment]
I've put the additional test for with empty second date and it works as expected but would be very appreciated for double check logic

File: packages/core/src/utilities/validation.js (hunk 1 above)
Comment (draperd) [reply packages/core/src/utilities/validation.js]: Thanks for adding these tests!
